### PR TITLE
Priority

### PR DESCRIPTION
**Pull request by cacizi41**

### CODE REVIEW

File: src/pqueue.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            self._sort_up()
+
+    def _sort_up(self):
+        idx = len(self.pqueue) - 1
+        while (idx-1)//2 >= 0:
+            if self.pqueue[idx][0] > self.pqueue[(idx-1)//2][0]:
+                temp_parent = self.pqueue[(idx-1)//2]
+                self.pqueue[(idx-1)//2] = self.pqueue[idx]
+                self.pqueue[idx] = temp_parent
+            idx = (idx-1)//2
```
Comment (joelstanner) [inline src/pqueue.py]: PEP8 E226 should be used for this class (whitespace around arithmetic operators). It has better readability:

```python
    def _sort_up(self):
        idx = len(self.pqueue) - 1
        while (idx - 1) // 2 >= 0:
            if self.pqueue[idx][0] > self.pqueue[(idx - 1) // 2][0]:
                temp_parent = self.pqueue[(idx - 1) // 2]
                self.pqueue[(idx - 1) // 2] = self.pqueue[idx]
                self.pqueue[idx] = temp_parent
            idx = (idx - 1) // 2
```

 Flake8 should be picking this up.

File: src/pqueue.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                comp_idx = 2*idx + 1
+            else:
+                if self.pqueue[2*idx + 1][0] > self.pqueue[2*idx + 2][0]:
+                    comp_idx = 2*idx + 1
+                else:
+                    comp_idx = 2*idx + 2
+            if self.pqueue[comp_idx][0] > self.pqueue[idx][0]:
+                temp_parent = self.pqueue[idx]
+                self.pqueue[idx] = self.pqueue[comp_idx]
+                temp_parent = self.pqueue[comp_idx]
```
Comment (joelstanner) [inline src/pqueue.py]: `temp_parent` doesn't seem to be doing anything here. You're assigning it twice, but it's not being used anywhere or returned.

File: src/test_pqueue.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+@pytest.mark.parametrize('value, result', POP_TABLE)
+def test_pop_pqueue(value, result):
+    from pqueue import Pqueue
+    new_pqueue = Pqueue(value)
+    assert new_pqueue.pop() == result[0]
+    assert new_pqueue.pop() == result[1]
+    assert new_pqueue.pop() == result[2]
+
```
Comment (joelstanner) [inline src/test_pqueue.py]: Be consistent with spacing between functions. There is only one here, but two up above.

File: src/test_pqueue.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@pytest.mark.parametrize('value, result', PEEK_TABLE)
+def test_peek_pqueue(value, result):
+    from pqueue import Pqueue
+    new_pqueue = Pqueue(value)
+    assert new_pqueue.peek() == result[0]
+
+
```
Comment (joelstanner) [inline src/test_pqueue.py]: Avoid extraneous whitespace at the end of files.

Comment (joelstanner) [inline src/test_pqueue.py]: This could use more testing. Make sure to include edge cases for your methods. What happens when you pop from an empty queue? A queue with 1 item? Peek an empty queue? Does everything function as expected? Use tests to prove that it does.

Comment (joelstanner) [inline src/test_pqueue.py]: Also - what happens when you enter a negative number for your priority? You can handle this either by explicitly documenting that your don't allow negative numbers, or you can handle it programmatically. Either way, your queue acts strangely when the priority goes negative.

**Comment by joelstanner** [comment]
Make an attempt to push atomic commits with concise and descriptive commit messages. "peek()docstring" is one thing in that commit, but there is a lot more going on in there.

File: src/pqueue.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                self.pqueue[comp_idx] = self.pqueue[idx]
+            idx = comp_idx
+
+
+
+
```
Comment (joelstanner) [inline src/pqueue.py]: Avoid all of this extra whitespace.
